compiler/rustc_feature/src/unstable.rs

```diff
@@ -529,6 +529,8 @@ declare_features! (
     (unstable, more_qualified_paths, "1.54.0", Some(86935)),
     /// Allows the `#[must_not_suspend]` attribute.
     (unstable, must_not_suspend, "1.57.0", Some(83310)),
+    /// Allows `mut ref` and `mut ref mut` identifier patterns.
+    (unstable, mut_ref, "CURRENT_RUSTC_VERSION", Some(123076)),
```

Comment:
I'm not sure what the procedure is for an edition feature, but if we follow the ["experimental feature gate"](https://lang-team.rust-lang.org/how_to/experiment.html) process, the feature gate needs to be marked as incomplete until t-lang approves the design.